public/javascripts/hello.js

```diff
@@ -1,3 +1,97 @@
 const es6Hello = "Hello OpenSD";
 
 console.log(es6Hello);
+
+var repoElements = Array.from(document.querySelectorAll("div.repository-selection button"));
+repoElements.forEach(function(el, i) {
+  el.addEventListener("click", function() {
+    el.classList.toggle("active");
+  });
+});
+
+var searchListeners = Array.from(document.querySelectorAll("div.main-search"));
+searchListeners.forEach(function(el, i){
+  el.addEventListener("click", function(){
+  //add button
+    if(event.target.classList.contains('add-search-button')){
+      var searchBarNodeClone = document.querySelectorAll('div.reusable-search')[0].cloneNode(true);
+      searchBarNodeClone.childNodes[0].childNodes[0].innerHTML = "Select " + "<span class=\"caret\"></span>";
+      searchBarNodeClone.childNodes[1].value="";
+      searchBarNodeClone.childNodes[1].placeholder="";
+      event.target.parentNode.parentNode.parentNode.insertBefore(searchBarNodeClone,event.target.parentNode.parentNode.nextSibling);
```

Comment:
No since these listeners are added dynamically and the context where they are added is dependent on the element an ID would be no benefit, as it would need to be added dynamically then stored somewhere for reference, which provides more overhead than actually necessary